pkg/cloud/vsphere/provisioner/govmomi/create.go

```diff
@@ -224,6 +217,30 @@ func (pv *Provisioner) cloneVirtualMachine(s *SessionContext, cluster *clusterv1
 		return fmt.Errorf("error fetching virtual machine or template properties: %s", err)
 	}
 
+	host, err := src.HostSystem(ctx)
+	if err != nil {
+		return fmt.Errorf("HostSystem failed. err=%s", err)
+	}
+
+	pool, err := s.finder.ResourcePoolOrDefault(ctx, machineConfig.MachineSpec.ResourcePool)
+	if err != nil {
+		klog.Warningf("Failed to find ResourcePool=%s err=%s. Attempting to create it.", machineConfig.MachineSpec.ResourcePool, err)
+
+		poolRoot, errRoot := host.ResourcePool(ctx)
```

Comment:
I think the finder has a function to get the resource pool so you don't even need to get the host, but this works too.

Reply:
Actually, I should be able to also call the root RP from the template itself. Testing it out now.

Reply:
Yea, I just looked at doing it that way, but this seems to work without a bunch of additional code and handles the standalone ESX host case using the same code path.